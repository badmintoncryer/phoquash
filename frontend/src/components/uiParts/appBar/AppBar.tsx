import { useState } from "react";
import AppBar from "@mui/material/AppBar";
import Box from "@mui/material/Box";
import Toolbar from "@mui/material/Toolbar";
import Typography from "@mui/material/Typography";
import IconButton from "@mui/material/IconButton";
import Menu from "@mui/material/Menu";
import MenuItem from "@mui/material/MenuItem";
import MenuIcon from "@mui/icons-material/Menu";
import { useNavigate } from "react-router-dom";
import { Grid } from "@mui/material";
import Logo from "../../../static/logo_transparent.png";

export default function ButtonAppBar() {
  const [anchorEl, setAnchorEl] = useState<null | HTMLElement>(null);
  const open = Boolean(anchorEl);
  const handleClick = (event: React.MouseEvent<HTMLButtonElement>) => {
    setAnchorEl(event.currentTarget);
  };
  const handleClose = () => {
    setAnchorEl(null);
  };
  const navigate = useNavigate();
  // HOMEページに遷移
  const handleToTopPage = () => {
    setAnchorEl(null);
    navigate("/");
  };
  const handleCreate = () => {
    setAnchorEl(null);
    navigate("/create");
  };

  return (
    <Box sx={{ flexGrow: 1 }}>
      <AppBar position="static">
        <Toolbar>
          <Grid container direction="row" alignItems="center">
            <Grid item>
              <IconButton
                size="large"
                edge="start"
                color="inherit"
                aria-label="menu"
                sx={{ mr: 2 }}
                aria-controls={open ? "basic-menu" : undefined}
                aria-haspopup="true"
                aria-expanded={open ? "true" : undefined}
                onClick={handleClick}
              >
                <MenuIcon />
              </IconButton>
              <Menu
                id="basic-menu"
                anchorEl={anchorEl}
                open={open}
                onClose={handleClose}
                MenuListProps={{
                  "aria-labelledby": "basic-butoon",
                }}
              >
                <MenuItem onClick={handleToTopPage}>HOMEへ戻る</MenuItem>
                <MenuItem onClick={handleCreate}>新規作成</MenuItem>
                <MenuItem disabled onClick={handleToTopPage}>
                  管理画面
                </MenuItem>
                <MenuItem disabled onClick={handleToTopPage}>
                  ログアウト
                </MenuItem>
              </Menu>
            </Grid>
            <Grid item>
              <img src={Logo} alt="logo" width={50} />
            </Grid>
            <Grid item>
              <Typography variant="h6" component="div" sx={{ flexGrow: 1 }}>
                phoquash
              </Typography>
            </Grid>
          </Grid>
        </Toolbar>
      </AppBar>
    </Box>
  );
}
