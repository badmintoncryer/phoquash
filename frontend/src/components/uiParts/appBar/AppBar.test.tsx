import React from 'react'
import { render, screen } from '@testing-library/react'
import { BrowserRouter } from 'react-router-dom'
import userEvent from '@testing-library/user-event'
import AppBar from './AppBar'

describe('AppBar componentのテスト', () => {
  test('AppBar componentのテスト', async () => {
    const { asFragment } = render(
      <BrowserRouter>
        <AppBar />
      </BrowserRouter>
    )
    expect(asFragment()).toMatchSnapshot()

    // 文字表示
    screen.getByText(/phoquash/i)
    // メニューボタンを押せるかテスト
    const button = screen.getByRole('button')
    await userEvent.click(button)
  })
})
