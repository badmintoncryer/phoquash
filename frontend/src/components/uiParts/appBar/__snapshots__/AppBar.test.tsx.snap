// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppBar componentのテスト AppBar componentのテスト 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-i9gxme"
  >
    <header
      class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionStatic css-hip9hq-MuiPaper-root-MuiAppBar-root"
    >
      <div
        class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-hyum1k-MuiToolbar-root"
      >
        <div
          class="MuiGrid-root MuiGrid-container css-1vam7s3-MuiGrid-root"
        >
          <div
            class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
          >
            <button
              aria-haspopup="true"
              aria-label="menu"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-edgeStart MuiIconButton-sizeLarge css-ancrnh-MuiButtonBase-root-MuiIconButton-root"
              tabindex="0"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="MenuIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
          >
            <img
              alt="logo"
              src="logo_transparent.png"
              width="50"
            />
          </div>
          <div
            class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
          >
            <div
              class="MuiTypography-root MuiTypography-h6 css-1juivf6-MuiTypography-root"
            >
              phoquash
            </div>
          </div>
        </div>
      </div>
    </header>
  </div>
</DocumentFragment>
`;
